internal/hotreload/hotreloadmodule.go

```diff
@@ -40,15 +42,21 @@ func (w hotReloadModule) VersionedFS(rel string, observeChanges bool) compute.Co
 type observerSink struct {
 	sink wsremote.Sink
 
-	triggerFullRebuildPredicate func(filepath string) bool
+	eventProcessor func(*wscontents.FileEvent) *wscontents.FileEvent
 }
 
 func (obs observerSink) Deposit(ctx context.Context, events []*wscontents.FileEvent) (bool, error) {
+	processedEvents := []*wscontents.FileEvent{}
 	for _, ev := range events {
-		if obs.triggerFullRebuildPredicate != nil && obs.triggerFullRebuildPredicate(ev.Path) {
-			return false, nil
+		processedEvent := ev
+		if obs.eventProcessor != nil {
+			processedEvent = obs.eventProcessor(ev)
```

Comment:
We need a comment here on why this API is necessary, for example: "Give an implementation the opportunity to rewrite the paths in event, to make them relative to a different directory."

I think we also need to have a signal on whether an event should skipped, or we should stop altogether. returning false here means stop all events.

I think something like the following is simpler:

```
if obs.eventProcessor == nil {
  return obs.sink.Deposit(ctx, events)
}

var transformed []*wscontents.FileEvent
for _, ev := range events {
  newEvent, continueProcessing := obs.eventProcessor(ev)
  if !continueProcessing {
    // processor signalled that no further events should be delivered.
    return false, nil
  }

  if newEvent != nil {
    transformed = append(transformed, newEvent)
  }
}

return obs.sink.Deposit(ctx, transformed)
```

Reply:
`whether an event should skipped, or we should stop altogether` is there a use-case for such an API at the moment?